lib/client.js

```diff
@@ -187,20 +187,17 @@ function _connect (client) {
     .on('error', function (err) {
       this[kError] = err
     })
-    .on('end', function () {
-      this.destroy(new Error('other side closed'))
```

Comment:
This won't cause `'error'` until nextTick, which would be after `finished(...)` which could invoke already at `'end'`.